### PR TITLE
drivers: display: Added Sharp memory display drivers.

### PR DESCRIPTION
**Pull request by rgujju**
Added support for sharp memory displays of the series LS0XX.

This driver supports following displays
LS012B7DD01 
LS012B7DD06 
LS013B7DH03 -> Tested
LS013B7DH05 
LS013B7DH06 
LS027B7DH01A 
LS032B7DD02
LS044Q7DH01

The following devicetree configuration is an example:

```
 &spi1 {
     compatible = "nordic,nrf-spim";
     status = "okay";
     sck-pin = <13>;
     mosi-pin = <15>;
     miso-pin = <24>;
 
     ls0xx@0 {
         compatible = "sharp,ls0xx";
         label = "LS0XX";
         spi-max-frequency = <2000000>;
         reg = <0>;
         width = <128>;
         height = <128>;
         scs-gpios = <&gpio0 20 GPIO_ACTIVE_HIGH>;
         extcomin-gpios = <&gpio0 22 GPIO_ACTIVE_HIGH>;
         disp-gpios = <&gpio0 29 GPIO_ACTIVE_HIGH>;
     };
 };
```
extcomin-gpios and disp-gpios are optional.
- If `disp-gpios` is provided then that GPIO (here GPIO0 29) Will be set to high during initialization and can be controlled using the blanking APIs.
- If `extcomin-gpios` is provided and `LS0XX_EXTMODE_SERIAL=n` then EXTCOMIN pin will be used for toggling VCOM.

The following LVGL options work for me

```
 CONFIG_DISPLAY=y
 CONFIG_SPI=y
 CONFIG_LS0XX=y
 
 CONFIG_LVGL_COLOR_DEPTH_1=y
 CONFIG_LVGL_BITS_PER_PIXEL=1
 CONFIG_LVGL_HOR_RES=128
 CONFIG_LVGL_VER_RES=128
 CONFIG_LVGL_DPI=150
 CONFIG_LVGL_VDB_SIZE=16
```

### CODE REVIEW

File: drivers/display/ls0xx.c [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+
+#include <string.h>
+#include <device.h>
+#include <drivers/display.h>
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <sys/byteorder.h>
+
+#include <display/cfb.h>
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:21]: Please remove #include <display/cfb.h>

File: drivers/display/ls0xx.c [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+ * -> high/1 means white, low/0 means black
+ * -> Display expects LSB first
+ * -> If devicetree contains DISP pin then it will be set to high
+ *      during initialization.
+ * -> VCOM is toggled through software by default (CONFIG_LS0XX_EXTMODE_SERIAL=y)
+ *    EXTMODE pin is connected to VSS 
+ * -> To control VCOM using external signal EXTCOMIN, set
+ *      CONFIG_LS0XX_EXTMODE_SERIAL=n
+ *      EXTMODE pin is connected to VDD
+ * -> When designing with a memory LCD panel, a decision must be made as to how
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:37]: Seem to be 1:1 copy or the VCOM chapter from the appn, please remove it and add short description + reference to the appn/chapter.

File: drivers/display/ls0xx.c [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+ *      Source:
+ * https://www.sharpsde.com/fileadmin/products/Displays/2016_SDE_App_Note_for_Memory_LCD_programming_V1.3.pdf
+ * -> Frame buf format for each row, This format is for LS0XX_PANEL_HEIGHT
+ *      Adding 2 to frame_buf for the line number and dummy byte
+ * +-------------------+-------------------+----------------+
+ * | line num (8 bits) | data (WIDTH bits) | dummy (8 bits) |
+ * +-------------------+-------------------+----------------+
+ */
+
+#define LS0XX_SPI_FREQ            DT_INST_PROP(0, spi_max_frequency)
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:66]: This define, how it is used now, seems not to be necessary, you can use DT_INST_PROP(0, spi_max_frequency) directly in the L428. Please check the others as well.

File: drivers/display/ls0xx.c [hunk 4]
```diff
@@ -0,0 +1,493 @@
...
+    int err;
+    memset(frame_buf, 0xFF, LS0XX_PANEL_HEIGHT*(((LS0XX_PANEL_WIDTH/LS0XX_PIXELS_PER_BYTE) + 2)));
+
+    uint8_t clear_cmd[2] = {LS0XX_BIT_CLEAR | LS0XX_EXTCOMIN_BIT, 0};
+    toggle_vcom_bit(driver);
+    buf[0].len = sizeof(uint16_t);
+    buf[0].buf = &clear_cmd;
+    buf_set.count = 1;
+
+    gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 1);
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:195]: LS0XX_SCS_PIN (serial chip select) should be handled by SPI driver

File: drivers/display/ls0xx.c [hunk 5]
```diff
@@ -0,0 +1,493 @@
...
+ * @details Clear the display memory and the frame_buf
+ *          setting it to all white
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_clear(const struct device *dev) {
+
+    struct ls0xx_data *driver = dev->data;
+    int err;
+    memset(frame_buf, 0xFF, LS0XX_PANEL_HEIGHT*(((LS0XX_PANEL_WIDTH/LS0XX_PIXELS_PER_BYTE) + 2)));
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:187]: `memset(frame_buf, 0xFF, sizeof(frame_buf));` ?

File: drivers/display/ls0xx.c [hunk 6]
```diff
@@ -0,0 +1,493 @@
...
+    struct ls0xx_data *driver = dev->data;
+    int err;
+    uint8_t write_cmd;
+
+    LOG_DBG("Updating lines from %d to %d", start_line, start_line+num_lines-1);
+
+    gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 1);
+
+    /* Send write command along with extcomin_bit */
+    write_cmd = LS0XX_BIT_WRITECMD | LS0XX_EXTCOMIN_BIT;
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:227]: Add ls0xx_spi_write_cmd function.

File: drivers/display/ls0xx.c [hunk 7]
```diff
@@ -0,0 +1,493 @@
...
+ * @brief Toggle VCOM to avoid DC bias
+ *
+ * @details This API will toggle either the EXTCOMIN pin
+ *          or the M1 bit in serial comm to toggle VCOM
+ *
+ * @param driver Pointer to the display driver
+ */
+static inline void toggle_vcom_bit(struct ls0xx_data *driver) {
+#if defined(CONFIG_LS0XX_EXTMODE_SERIAL)
+    extcomin_bit = extcomin_bit ? 0x00 : LS0XX_BIT_VCOM;
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:133]: Please follow https://docs.zephyrproject.org/latest/contribute/index.html?highlight=coding%20style#coding-style

**Review by jfischer-no — CHANGES_REQUESTED** [review]

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Please add shield support for the display based on e.g. LS013B7DH05, like https://www.adafruit.com/product/3502 or https://www.waveshare.com/1.3inch-memory-lcd.htm

File: drivers/display/ls0xx.c (hunk 6 above)
Comment (rgujju) [reply drivers/display/ls0xx.c:227]: Do you mean a separate SPI transfer only for the write cmd? Or a separate function just to populate the buf[0] ?

File: drivers/display/ls0xx.c (hunk 4 above)
Comment (rgujju) [reply drivers/display/ls0xx.c:195]: 1) This display has an inverted SCS. It is active high. Does the SPI driver support this?
2) I get the following ASSERT when trying to use the SPI driver
```
 /* Validate CS active levels are equivalent */
__ASSERT(spi_context_cs_active_level(ctx) ==
 (ctx->config->cs->gpio_dt_flags & GPIO_ACTIVE_LOW)
```

**Review by vanwinkeljan — CHANGES_REQUESTED** [review]

File: drivers/display/ls0xx.c [hunk 8]
```diff
@@ -0,0 +1,496 @@
...
+
+#define LS0XX_PIXELS_PER_BYTE     8U
+/* Adding 2 for the line number and dummy byte */
+#define LS0XX_BYTES_PER_LINE      ((LS0XX_PANEL_WIDTH/LS0XX_PIXELS_PER_BYTE)+2)
+
+#define LS0XX_BIT_WRITECMD        0x01
+#define LS0XX_BIT_VCOM            0x02
+#define LS0XX_BIT_CLEAR           0x04
+
+static uint8_t frame_buf[LS0XX_PANEL_HEIGHT*(((LS0XX_PANEL_WIDTH/LS0XX_PIXELS_PER_BYTE)+2))];
```
[the remark this answers is not in the document]
Comment (vanwinkeljan) [reply drivers/display/ls0xx.c:78]: Yes that was the idea

@jfischer-phytec-iot any opinion about the idea?

File: drivers/display/ls0xx.c (hunk 4 above)
Comment (jfischer-phytec-iot) [reply drivers/display/ls0xx.c]: Yes SPI driver API supports it and it is mandatory to describe it, for example https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/shields/waveshare_epaper/waveshare_epaper_gdeh0154a07.overlay#L9

File: drivers/display/ls0xx.c (hunk 8 above)
[the remark this answers is not in the document]
Comment (jfischer-phytec-iot) [reply drivers/display/ls0xx.c:78]: TBH I need to see the code :-) but sounds good.

File: drivers/display/ls0xx.c [hunk 9]
```diff
@@ -0,0 +1,496 @@
...
+ *      toggle VCOM and return
+ * -> high/1 means white, low/0 means black
+ * -> Display expects LSB first
+ * -> If devicetree contains DISP pin then it will be set to high
+ *      during initialization.
+ * -> Sharp memory displays require toggling the VCOM signal periodically
+ *      to prevent a DC bias ocurring in the panel. The DC bias can damage
+ *      the LCD and reduce the life. This signal must be supplied
+ *      externally from either serial input or an external signal on
+ *      the EXTCOMIN pin.
```
[the remark this answers is not in the document]
Comment (jfischer-phytec-iot) [reply drivers/display/ls0xx.c:38]: An other use case for #23075?

File: boards/shields/ls0xx_generic/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,88 @@
+.. _ls0xx_generic_shield:
```
Comment (jfischer-phytec-iot) [inline boards/shields/ls0xx_generic/doc/index.rst:1]: Thank you for the shield support. Can you move shield support to own commit, please?

File: drivers/display/ls0xx.c (hunk 9 above)
[the remark this answers is not in the document]
Comment (rgujju) [reply drivers/display/ls0xx.c:38]: @vanwinkeljan , using a k_timer is becoming very complicated. To avoid contention on spi bus i am trying to use a mutex. but since the expiry_fn of k_timer is called in an isr i am not able to use a mutex. So i might have to use a workqueue. 

Alternatively i suggest using a K_THREAD. What do you think?

[the remark this answers is not in the document]
Comment (vanwinkeljan) [reply drivers/display/ls0xx.c:38]: > Alternatively i suggest using a K_THREAD. What do you think?

Sounds like a reasonable alternative


@jfischer-phytec-iot 
 > An other use case for #23075?

Combined with the queue manager mentioned in that issues this driver sounds like a perfect candidate.

[the remark this answers is not in the document]
Comment (rgujju) [reply drivers/display/ls0xx.c:38]: Hello @vanwinkeljan ,

This k_thread/k_timer (any asynchronous method really, asynchronous to the write command) approach has one flaw, which may or may not need to be addressed.
If the user is using the same SPI interface for this display and some other sensor. Then there might be contention during a sensor read/write and VCOM toggle command over SPI.

To resolve this, we can either 
1) limit the user to not using the same SPI interface by adding a note in the shields documentation.
2) Use the current approach of using the generic write API if the user selects LS0XX_EXTMODE_SERIAL=y and use a thread for GPIO toggling if LS0XX_EXTMODE_SERIAL=n.

**Comment by rgujju** [comment]
Hi @jfischer-phytec-iot, @vanwinkeljan ,

I have incorporated most of your comments. let me know what you think. 
Few questions:
1) What should be the priority of the vcom toggling thread. currently set to 3.
2) References section of the shield documentation does not get converted to html. is there something wrong with the formatting?
3) I still get an assert when i try to use the SPI driver for SCS with GPIO_ACTIVE_HIGH. Do you know of any existing driver which has a GPIO_ACTIVE_HIGH chip select I can refer?

File: drivers/display/ls0xx.c (hunk 9 above)
[the remark this answers is not in the document]
Comment (vanwinkeljan) [reply drivers/display/ls0xx.c]: @rgujju Options 2 sounds the best solution until we have something like the async sequence manager

[the remark this answers is not in the document]
Comment (rgujju) [reply drivers/display/ls0xx.c]: yes! I have gone ahead with option 2 in the latest commits.

**Comment by vanwinkeljan** [comment]
Thanks for the update, it looks ok for me.

> 1. What should be the priority of the vcom toggling thread. currently set to 3.

Not sure which prio to set, one option is to make it configurable via Kconfig and set it 3 by default

> 2. References section of the shield documentation does not get converted to html. is there something wrong with the formatting?

I think the links only show up if they are referenced somewhere in the rst file



@jfischer-phytec-iot could you have a second look?

**Comment by jfischer-phytec-iot** [comment]
> @jfischer-phytec-iot could you have a second look?

I got LS013B7DH05 and will have a closer look at the driver in the next days.

File: drivers/display/ls0xx.c [hunk 11]
```diff
@@ -0,0 +1,501 @@
...
+ * +-------------------+-------------------+----------------+
+ * | line num (8 bits) | data (WIDTH bits) | dummy (8 bits) |
+ * +-------------------+-------------------+----------------+
+ */
+
+#define LS0XX_SCS_PIN             DT_INST_GPIO_PIN(0, scs_gpios)
+#define LS0XX_SCS_CNTRL           DT_INST_GPIO_LABEL(0, scs_gpios)
+
+#if DT_INST_NODE_HAS_PROP(0, disp_gpios)
+	#define LS0XX_DISP_PIN        DT_INST_GPIO_PIN(0, disp_gpios)
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:64]: Do not indent `#define`

File: drivers/display/ls0xx.c [hunk 12]
```diff
@@ -0,0 +1,501 @@
...
+		#define LS0XX_EXTCOMIN_CNTRL  DT_INST_GPIO_LABEL(0, extcomin_gpios)
+	#endif
+#endif
+
+#define LS0XX_PANEL_WIDTH         DT_INST_PROP(0, width)
+#define LS0XX_PANEL_HEIGHT        DT_INST_PROP(0, height)
+
+#define LS0XX_PIXELS_PER_BYTE     8U
+/* Adding 2 for the line number and dummy byte */
+#define LS0XX_BYTES_PER_LINE      ((LS0XX_PANEL_WIDTH/LS0XX_PIXELS_PER_BYTE)+2)
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:80]: ) + 2)

File: drivers/display/ls0xx.c [hunk 13]
```diff
@@ -0,0 +1,501 @@
...
+#define LS0XX_BIT_CLEAR           0x04
+
+static uint8_t line_buf[LS0XX_BYTES_PER_LINE];
+
+static struct spi_buf tx_buf;
+static struct spi_buf_set buf_set = {.buffers = &tx_buf, .count = 1};
+
+struct ls0xx_data {
+	struct device *scs;
+#if defined(LS0XX_DISP_CNTRL)
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:93]: Do not obfuscate where it comes from, change to #if DT_INST_NODE_HAS_PROP(0, disp_gpios), the same below.

I also thing we should reconsider this practice, sacrifice these few bytes and remove all _#if_ and use` IS_ENABLED`.

File: drivers/display/ls0xx.c [hunk 14]
```diff
@@ -0,0 +1,501 @@
...
+	struct device *extcomin;
+#endif
+	struct device *spi_dev;
+	struct spi_config spi_config;
+};
+static uint8_t extcomin_bit;
+
+#if defined(CONFIG_LS0XX_VCOM_DRIVER)
+
+#if defined(CONFIG_LS0XX_VCOM_EXTERNAL)
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:106]: Please remove LS0XX_VCOM_EXTERNAL and rework this macro nightmare.

File: drivers/display/ls0xx.c [hunk 15]
```diff
@@ -0,0 +1,501 @@
...
+extcomin-gpios is not present in devicetree"
+#endif /* LS0XX_EXTCOMIN_CNTRL */
+#define LS0XX_EXTCOMIN_BIT 0x00
+#else
+#define LS0XX_EXTCOMIN_BIT extcomin_bit
+#endif /* CONFIG_LS0XX_VCOM_EXTERNAL */
+
+#if defined(CONFIG_LS0XX_VCOM_EXTERNAL)
+/**
+ * @brief Toggle VCOM to avoid DC bias
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:118]: I do not see any reasons to document a static function this way.

File: drivers/display/ls0xx.c [hunk 16]
```diff
@@ -0,0 +1,501 @@
...
+ *
+ * @details This API will toggle either the EXTCOMIN pin
+ *          or the M1 bit in serial comm to toggle VCOM
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:121]: How is M1 bit toggled periodical?

File: drivers/display/ls0xx.c [hunk 17]
```diff
@@ -0,0 +1,501 @@
...
+ *
+ * @param driver Pointer to the display driver
+ */
+static void ls0xx_vcom_toggle(void *a, void *b, void *c)
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:125]: Why can not you use a workqueue thread and trigger it from a timer?

File: drivers/display/ls0xx.c [hunk 18]
```diff
@@ -0,0 +1,501 @@
...
+	}
+}
+
+/* Driver will handle VCOM toggling */
+K_THREAD_STACK_DEFINE(vcom_toggle_stack, 256);
+struct k_thread vcom_toggle_thread;
+#endif /* CONFIG_LS0XX_VCOM_EXTERNAL */
+
+#else
+#define LS0XX_EXTCOMIN_BIT 0x00
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:143]: Third place where LS0XX_EXTCOMIN_BIT is defined, please rework

File: drivers/display/ls0xx.c [hunk 19]
```diff
@@ -0,0 +1,501 @@
...
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_blanking_off(const struct device *dev)
+{
+#if defined(LS0XX_DISP_CNTRL)
+	struct ls0xx_data *driver = dev->data;
+
+	return gpio_pin_set(driver->disp, LS0XX_DISP_PIN, 1);
+#else
+	LOG_ERR("Unsupported");
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:161]: LOG_WRN would be more appropriate, same below for ls0xx_blanking_on

File: drivers/display/ls0xx.c [hunk 20]
```diff
@@ -0,0 +1,501 @@
...
+ *          starting from start_line.
+ *
+ * @param dev Pointer to the display device
+ * @param start_line Starting line number, starts with 1
+ *                   corresponds to Y co-ordinate + 1
+ * @param num_lines Number of lines ie height to update
+ * @param buf  Buffer containing data to write
+ */
+#if defined(__GNUC__)
+	__attribute__((optimize("O3")))
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:223]: Remove this.

File: drivers/display/ls0xx.c [hunk 21]
```diff
@@ -0,0 +1,501 @@
...
+#endif
+static int ls0xx_update_display(const struct device *dev,
+								uint16_t start_line,
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:226]: Fix indent here and below. (settings of your editor are probably wrong.)

File: drivers/display/ls0xx.c [hunk 22]
```diff
@@ -0,0 +1,501 @@
...
+{
+	struct ls0xx_data *driver = dev->data;
+
+	int err;
+	uint8_t write_cmd;
+	uint16_t currentline;
+
+	LOG_DBG("Lines %d to %d", start_line, start_line+num_lines-1);
+
+	gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 1);
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:238]: Why is this ignores CS handling of SPI controller driver?

File: drivers/display/ls0xx.c [hunk 23]
```diff
@@ -0,0 +1,501 @@
...
+
+	write_cmd = LS0XX_BIT_WRITECMD | LS0XX_EXTCOMIN_BIT;
+	extcomin_bit = LS0XX_EXTCOMIN_BIT ? 0x00 : LS0XX_BIT_VCOM;
+	tx_buf.len = sizeof(uint8_t);
+	tx_buf.buf =  &write_cmd;
+	err = spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+
+	if (buf != NULL) {
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:246]: just error and exit at the top if no buffer is provided.

File: drivers/display/ls0xx.c [hunk 24]
```diff
@@ -0,0 +1,501 @@
...
+		/* Send each line to the screen including
+		 * the line number and dummy bits
+		 */
+		tx_buf.len = sizeof(uint8_t)*LS0XX_BYTES_PER_LINE;
+		tx_buf.buf = &line_buf[0];
+		for (currentline = start_line; currentline <= start_line+num_lines-1; currentline++) {
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:252]: `for (int l = start_line; l <= max_line; l++)`

File: drivers/display/ls0xx.c [hunk 25]
```diff
@@ -0,0 +1,501 @@
...
+ *
+ * @param driver Pointer to the display driver
+ */
+static void ls0xx_vcom_toggle(void *a, void *b, void *c)
+{
+	struct ls0xx_data *driver = (struct ls0xx_data *)a;
+
+	while (1) {
+		gpio_pin_toggle(driver->extcomin, LS0XX_EXTCOMIN_PIN);
+		k_msleep(10);
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:131]: why 10 and 500?

File: drivers/display/ls0xx.c [hunk 26]
```diff
@@ -0,0 +1,501 @@
...
+
+static uint8_t line_buf[LS0XX_BYTES_PER_LINE];
+
+static struct spi_buf tx_buf;
+static struct spi_buf_set buf_set = {.buffers = &tx_buf, .count = 1};
+
+struct ls0xx_data {
+	struct device *scs;
+#if defined(LS0XX_DISP_CNTRL)
+	struct device *disp;
```
Comment (jfischer-phytec-iot) [inline drivers/display/ls0xx.c:94]: better pin_disp_dev or pin_disp_ctrl

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: drivers/display/ls0xx.c (hunk 22 above)
Comment (rgujju) [reply drivers/display/ls0xx.c:238]: I could not find any driver that uses SPI driver with active high CS. When I try it get the following assert which leads me to think it the driver does not support it. Do you know any code i code refer to which uses active high CS. 
```
 /* Validate CS active levels are equivalent */
 __ASSERT(spi_context_cs_active_level(ctx) ==
  (ctx->config->cs->gpio_dt_flags & GPIO_ACTIVE_LOW),
  "Devicetree and spi_context CS levels are not equal");
```

File: drivers/display/ls0xx.c (hunk 23 above)
Comment (rgujju) [reply drivers/display/ls0xx.c:246]: We need to send the VCOM toggle command

File: drivers/display/ls0xx.c (hunk 25 above)
Comment (rgujju) [reply drivers/display/ls0xx.c:131]: It just needs to be a pulse which is periodic atleast every second.

File: drivers/display/ls0xx.c (hunk 16 above)
Comment (rgujju) [reply drivers/display/ls0xx.c:121]: Will remove this. M1 bit is toggled in the write API now.

File: drivers/display/ls0xx.c (hunk 17 above)
Comment (rgujju) [reply drivers/display/ls0xx.c:125]: we could. but i thought it would be complicated for a simple driver like this.

**Comment by nick-kraus** [comment]
@rgujju Are you still moving forward with this PR? If not I will probably build off the work done here and try to get this support merged in, would be great to have support for these displays.

**Comment by rgujju** [comment]
@nick-kraus Ill try to merge this in a week or two if thats ok?
